pysal/contrib/viz/plot.py

```diff
@@ -0,0 +1,47 @@
+"""
+Moran Plot using PySAL and Matplotlib
+
+"""
+
+__author__ = "Marynia Kolak <marynia.kolak@gmail.com>"
+
+import pandas as pd
+import numpy as np
+import pysal as ps
+import matplotlib.pyplot as plt
+
+
+def Moran(var, w, **kwargs):
+	'''
+    Produce basic Moran Plot 
+    ...
+
+    Arguments
+    ---------
+
+    var	         	: array
+                      values of variable
+    w       	    : array
+                      values of spatial weight
+    '''
+	w.transform = 'r'
+	slag = ps.lag_spatial(w, var)
+
+	 # Z-Score standardisation
+	y_std   = (var - var.mean())/var.std()
+	yl_std  = (slag - slag.mean())/slag.std()
+
+	fig1 = plt.figure(figsize=(5,5))
+
+	plt.scatter(y_std, yl_std, s=60, color='k', alpha=.6)
+
```

Comment:
Definitely, the idea with this first pass is just to test the waters and
see what would work! It's still in progress -- you can see my notebook
kicking around some ideas and configs here
<https://github.com/Makosak/PySalViz/blob/master/Untitled%20Folder/Scatter%20Plots%20-%20Moran.ipynb>.
so:


   - *Module name*: renaming to 'canned' as a master would be fine -- i'll
   adjust that for the next round. (or 'cooked' ? ;P i guess there's time to
   consider options.)
   - *Default style: *yup, the default style is not terribly pretty in
   matplotlibs. I was trying to keep it to matplotlib for now following the
   viz project specs, and to reduce dependencies, but would it make sense to
   have a parallel style in seaborn? shouldn't be too difficult, and would add
   some options.. (Also, check out the 4-color matplotlib quadrant figure in
   the notebook above...)
   - *kwargs: *these I was reserving for title='', xlabel='', ylabel='',
   and figsize() following matplotlib logic -- see comments in the notebook
   link above. I may be totally misusing the term 'kwargs' here so apologies;
   still green.. I haven't added them yet in this version but just to get
   something out! These would be optional, but a way to enhance the visual if
   you want it. Maybe we could connect seaborn through that, for a more
   advanced user? like seaborn=false will give you a matplotlib default
   object, but if seaborn=true then we assume seaborn has been added as a
   dependency, and you get a nicer seaborn visual?
   - *m as input: *Serge had just recommended that too, as it also gets
   around issues of weight transformations. working on this tonight for the
   next iteration.
   - *adding lines: *Also working on that, for next iteration; probably
   envelopes too? or envelopes=true as an optional input to add them?



On Sun, Apr 24, 2016 at 3:26 AM, Dani Arribas-Bel <notifications@github.com>
wrote:

> In pysal/contrib/viz/plot.py
> <https://github.com/pysal/pysal/pull/797#discussion_r60842378>:
>
> > +    var	         	: array
> > +                      values of variable
> > +    w       	    : array
> > +                      values of spatial weight
> > +    '''
> > +	w.transform = 'r'
> > +	slag = ps.lag_spatial(w, var)
> > +
> > +	 # Z-Score standardisation
> > +	y_std   = (var - var.mean())/var.std()
> > +	yl_std  = (slag - slag.mean())/slag.std()
> > +
> > +	fig1 = plt.figure(figsize=(5,5))
> > +
> > +	plt.scatter(y_std, yl_std, s=60, color='k', alpha=.6)
> > +
>
> We might want to add a linear (and potentially non-linear) fit line. Also,
> maybe we could add a legend for that line that includes the coefficient for
> Moran's I (as it's its slope)?
>
> At the very least, I'd add the linear fit and maybe introduce an option
> for a LOESS line, although that'd make us depend on statsmodels.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pysal/pysal/pull/797/files/2fb3d3b7eaa9d6bb2032f7393d54f04478937eb6#r60842378>
>